data/uhc_pack/functions/pre_generation/stage/step/teleport/dynamic/teleport.mcfunction

```diff
@@ -0,0 +1,13 @@
+# Teleport the target marker to the desired location
+# There is no need to load any other chunks, since @s selected entities stay in memory and do not have to be reselected
+
+# Update storage with the new coordinates
+execute store result storage uhc_pack:dynamic_tp Pos[0] double 1 run scoreboard players get X uhcDynTP
```

Comment:
It appears that this command fails unless you have initialized the list to 3 elements ahead of time using:

```
data modify storage uhc_pack:dynamic_tp Pos set value [0d, 0d, 0d]
```

This should be done in the `create_storage` function